packages/neon/neon_talk/lib/src/widgets/unread_indicator.dart

```diff
@@ -0,0 +1,54 @@
+import 'package:flutter/material.dart';
+import 'package:nextcloud/spreed.dart' as spreed;
+
+/// Displays the number of unread messages and whether the user was mentioned directly or indirectly for a given [room].
+class TalkUnreadIndicator extends StatelessWidget {
+  /// Creates a new Talk unread indicator.
+  const TalkUnreadIndicator({
+    required this.room,
+    super.key,
+  });
+
+  /// The room that the indicator will display unread messages and mentions for.
+  final spreed.Room room;
+
+  @override
+  Widget build(BuildContext context) {
+    if (room.unreadMessages == 0) {
+      return const SizedBox.shrink();
+    }
+
+    final Color textColor;
+    final Color? backgroundColor;
+    final Color borderColor;
+
+    if (room.unreadMentionDirect || spreed.RoomType.fromValue(room.type).isSingleUser) {
+      textColor = Theme.of(context).colorScheme.onPrimary;
+      backgroundColor = Theme.of(context).colorScheme.primary;
+      borderColor = Theme.of(context).colorScheme.onBackground;
+    } else {
+      textColor = Theme.of(context).colorScheme.onBackground;
+      backgroundColor = null;
+      borderColor = room.unreadMention ? Theme.of(context).colorScheme.primary : Colors.grey;
```

Comment:
Onto the styling.
I think that using the primary color is a bit to much. I think we should use `primaryContainer`. It is currently not used by the nc theme but when the user uses the native m3 design it makes a big difference.

---

Also imo the different colors don't make clear that one involves a mention and one doesn't. We should maybe make mentions clearer by using an `@` sign.

This is how signal handles it (if only one unread message exists and it is a mention only the `@` symbol is shown). 
![signal-2024-02-25-201324](https://github.com/nextcloud/neon/assets/25266387/cb04e9cc-b8dd-472f-bd7c-45981ebb6987)

Reply:
I really like the `@`, I never noticed that Signal does this :sweat_smile: I just copied the behaviour of the Talk web client, but this is definitely much better.